pkg/agent/client/client.go

```diff
@@ -362,9 +362,9 @@ func (c *client) fetchBundles(ctx context.Context, federatedBundles []string) ([
 		case codes.OK:
 			bundles = append(bundles, bundle)
 		case codes.NotFound:
-			c.c.Log.WithError(err).WithField(telemetry.FederatedBundle, b).Warn("Federated bundle not found")
+			c.addErrorFields(err).WithField(telemetry.FederatedBundle, b).Warn("Federated bundle not found")
```

Comment:
Related to my suggestion below, if we check for `err == nil` in this new method, we can call it once above this `switch` block, rather than for each error log in the switch cases.

Reply:
It would be called once for each one of the `federatedBundles` that are being processed and use the logger that that method returns in each one of the cases, right?

Reply:
Right, if you declare the local logger with the error fields before the switch block within the for loop, it will only use that logger for the federated bundle currently being examined.

Reply:
Fixed at 55c88fa